common/src/main/java/com/flipkart/varadhi/utils/YamlLoader.java

```diff
@@ -1,16 +1,22 @@
 package com.flipkart.varadhi.utils;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import com.flipkart.varadhi.exceptions.VaradhiException;
-import org.yaml.snakeyaml.Yaml;
 
-import java.nio.file.Files;
-import java.nio.file.Path;
+import java.io.File;
 
 public class YamlLoader {
 
+    private static final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+
+    static {
+        mapper.findAndRegisterModules();
+    }
+
```

Comment:
Can you add a testcase which covers the generic use cases at nested level and top level.

Reply:
add the test cases
using snake yaml following failure happens:
```
java.lang.ClassCastException: class java.lang.String cannot be cast to class com.flipkart.varadhi.utils.YamlLoaderTest$MyEnum (java.lang.String is in module java.base of loader 'bootstrap'; com.flipkart.varadhi.utils.YamlLoaderTest$MyEnum is in unnamed module of loader 'app')
```

but with jackson test passes